# Config file for Joy Interface with the XBox360 Controller
###########################################################

# Device name
device: xbox
feedback: true
emergencyButtonIndex: 5

# Set a hard emergency stop with the button specified above?
hardEmergencyStop: false

# Mapping of the axes: each value stands for an index of the received axes; -1 for an inactive axis.
axisMapping: [0, 1, -1, 3, 4, -1]
# The Joy Manager interprets the axis as follows:
# [lateral, heading, roll, turning, vertical, pitch]

# Mapping of the buttons: each value stands for an index of the received buttons
buttonMapping: [0, 1, 2, 3, 4, 5, 6, 7]

# Intensity of the feedback [0.0, 1.0]
feedbackIntensity: 1.0

subscribers:
  joy:
    topic: joy
    queue_size: 10

publishers:
  anyJoy:
    # topic is specified as launch file parameter
    queue_size: 10
    latch: false
  feedback:
    topic: /feedback/xbox
    queue_size: 10
    latch: false

# Notfication Levels from any_msgs/Notification.msg
# LEVEL_FATAL = 4
# LEVEL_ERROR = 3
# LEVEL_WARN = 2
# LEVEL_INFO = 1
# LEVEL_DEBUG = 0
notification:
  level:
    screen: 1
    vibration: 3
  output_devices:
      operator_screen:
        topic: /notification
        queue_size: 10
